src/plugProjectOgawaU/ogScreen.cpp

```diff
@@ -1,1880 +1,461 @@
-    lbl_8051D530:
-        .4byte 0x3F4CCCCD
-        .4byte 0x00000000
-    .global lbl_8051D538
-    lbl_8051D538:
-        .4byte 0x3FF00000
-        .4byte 0x00000000
-    .global lbl_8051D540
-    lbl_8051D540:
-        .4byte 0x3FC90FDB
-    .global lbl_8051D544
-    lbl_8051D544:
-        .4byte 0x3E4CCCCC
-    .global lbl_8051D548
-    lbl_8051D548:
-        .4byte 0x40240000
-        .4byte 0x00000000
-    .global lbl_8051D550
-    lbl_8051D550:
-        .4byte 0x43300000
-        .4byte 0x80000000
-    .global lbl_8051D558
-    lbl_8051D558:
-        .4byte 0x437F0000
-        .4byte 0x00000000
-*/
+#include "trig.h"
+#include "System.h"
 
 namespace og {
-
 namespace Screen {
 
-/*
- * --INFO--
- * Address:	80301EB8
- * Size:	000008
- */
-void ArrowAlphaBlink::setSpeed(float)
-{
-	/*
-stfs     f1, 4(r3)
-blr
-	*/
-}
+ResTIMG* PikiIconTextureResTIMG[19];
 
-/*
- * --INFO--
- * Address:	........
- * Size:	000008
- */
-void ArrowAlphaBlink::setAmp(float)
+void _Print(char* format, ...)
 {
-	// UNUSED FUNCTION
+	OSReport("ogScreen");
+	char buffer[512];
+	va_list args;
+	va_start(args, format);
+	sprintf(buffer, format, __FILE__, buffer);
+	vprintf(buffer, args);
 }
 
 /*
  * --INFO--
- * Address:	........
+ * Address:	80301EB8
  * Size:	000008
  */
-void ArrowAlphaBlink::setBottom(float)
-{
-	// UNUSED FUNCTION
-}
+void ArrowAlphaBlink::setSpeed(float speed) { m_speed = speed; }
 
 /*
  * --INFO--
  * Address:	80301EC0
  * Size:	0000C0
  */
-void ArrowAlphaBlink::calc(void)
+float ArrowAlphaBlink::calc()
 {
-	/*
-stwu     r1, -0x20(r1)
-lfs      f3, lbl_8051D508@sda21(r2)
-lwz      r4, sys@sda21(r13)
-lfs      f1, 4(r3)
-lfs      f0, 0x54(r4)
-fmuls    f2, f1, f0
-lfs      f1, 0(r3)
-lfs      f0, lbl_8051D50C@sda21(r2)
-fmadds   f1, f3, f2, f1
-stfs     f1, 0(r3)
-lfs      f1, 0(r3)
-fcmpo    cr0, f1, f0
-ble      lbl_80301EFC
-fsubs    f0, f1, f0
-stfs     f0, 0(r3)
-
-lbl_80301EFC:
-lfs      f2, 0(r3)
-lfs      f0, lbl_8051D51C@sda21(r2)
-lfs      f3, 8(r3)
-fcmpo    cr0, f2, f0
-lfs      f1, lbl_8051D510@sda21(r2)
-bge      lbl_80301F40
-lfs      f0, lbl_8051D514@sda21(r2)
-lis      r4, sincosTable___5JMath@ha
-addi     r4, r4, sincosTable___5JMath@l
-fmuls    f0, f2, f0
-fctiwz   f0, f0
-stfd     f0, 8(r1)
-lwz      r0, 0xc(r1)
-rlwinm   r0, r0, 3, 0x12, 0x1c
-lfsx     f0, r4, r0
-fneg     f0, f0
-b        lbl_80301F64
-
-lbl_80301F40:
-lfs      f0, lbl_8051D518@sda21(r2)
-lis      r4, sincosTable___5JMath@ha
-addi     r4, r4, sincosTable___5JMath@l
-fmuls    f0, f2, f0
-fctiwz   f0, f0
-stfd     f0, 0x10(r1)
-lwz      r0, 0x14(r1)
-rlwinm   r0, r0, 3, 0x12, 0x1c
-lfsx     f0, r4, r0
-
-lbl_80301F64:
-fadds    f2, f1, f0
-lfs      f1, lbl_8051D520@sda21(r2)
-lfs      f0, 0xc(r3)
-fmuls    f2, f3, f2
-fmadds   f1, f2, f1, f0
-addi     r1, r1, 0x20
-blr
-	*/
+	_00 += 30.0f * (m_speed * sys->m_secondsPerFrame);
+	if (_00 > TAU) {
+		_00 = _00 - TAU;
+	}
+	float factor = _08 * (1.0f + pikmin2_sinf(_00));
+	float scale  = 0.5f;
```

Comment:
might this be a simple division by two?

Reply:
Looks like it: https://decomp.me/scratch/kNKXB

Reply:
will update o7